src/fidesops/service/connectors/sql_connector.py

```diff
@@ -183,6 +183,32 @@ def create_client(self) -> Engine:
             echo=not self.hide_parameters,
         )
 
+    # Overrides BaseConnector.cursor_result_to_rows
+    @staticmethod
+    def cursor_result_to_rows(results: LegacyCursorResult) -> List[Row]:
+        """Convert SQLAlchemy results to a list of dictionaries"""
+        columns: List[Column] = results.cursor.description
+        l = len(columns)
+        rows = []
+        for row_tuple in results:
+            rows.append({columns[i][0]: row_tuple[i] for i in range(l)})
+        return rows
+
+    def retrieve_data(
+        self, node: TraversalNode, policy: Policy, input_data: Dict[str, List[Any]]
+    ) -> List[Row]:
+        """Retrieve sql data"""
+        query_config = self.query_config(node)
+        client = self.client()
+        stmt: Optional[TextClause] = query_config.generate_query(input_data, policy)
+        if stmt is None:
+            return []
+        logger.info(f"Starting data retrieval for {node.address}")
+        with client.connect() as connection:
+            # fixme: update mssql type too to LegacyCursorResult
+            results: LegacyCursorResult = connection.execute(stmt)
+            return MySQLConnector.cursor_result_to_rows(results)
```

Comment:
Unless I'm missing something, this override seems to just call `MySQL.cursor_result_to_rows` instead of `SQLConnector.cursor_result_to_rows`. However, cursor_result_to_rows is a staticmethod and we can call that from the instance itself.

Instead of the overrides you've added here for MySQL and MSSQL `retrieve_data`, you should be able to get rid of it, and have the parent `SQLConnector.retrieve data` return `self.cursor_result_to_rows`.  When `MySQL.retrieve_data` is run, `self` will be an instance of the MySQL Connector, so the subclassed `cursor_result_to_rows` should be selected.